demo/rust/filedemo/src/main.rs

```diff
@@ -0,0 +1,195 @@
+/*
+    Copyright 2018-2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use clap::{App, Arg, ArgGroup};
+use itertools::Itertools;
+use pv_porcupine::{pv_keyword_paths, pv_library_path, pv_model_path, Porcupine, KEYWORDS};
+use rodio::{source::Source, Decoder};
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::time::SystemTime;
+
+fn porcupine_demo(
+    input_audio_path: PathBuf,
+    library_path: PathBuf,
+    model_path: PathBuf,
+    keyword_paths: Vec<PathBuf>,
+    keywords: Vec<String>,
+    sensitivities: Vec<f32>,
+) {
+    let soundfile = BufReader::new(File::open(input_audio_path).unwrap());
+    let audiosource = Decoder::new(soundfile).unwrap();
+
+    let porcupine = Porcupine::new(library_path, model_path, &keyword_paths, &sensitivities)
+        .expect("Failed to create Porcupine");
+
+    if porcupine.sample_rate() != audiosource.sample_rate() {
+        panic!(
+            "Audio file should have the expected sample rate of {}, got {}",
+            porcupine.sample_rate(),
+            audiosource.sample_rate()
+        );
+    }
+
+    for frame in &audiosource.chunks(porcupine.frame_length() as usize) {
+        let frame = frame.collect_vec();
+        if frame.len() == porcupine.frame_length() as usize {
+            let keyword_index = porcupine.process(&frame).unwrap();
+            if keyword_index >= 0 {
+                println!(
+                    "[{}] Detected {}",
+                    SystemTime::now()
+                        .duration_since(SystemTime::UNIX_EPOCH)
```

Comment:
Can you make the time more readable?

Reply:
For this actually, you may want to print when in the file it occurred, e.g. "Detected porcupine at 3:40", rather than the timestamp.